### DISCUSSION TITLE
开发一个新的插件时，如何使用E2E测试框架

### DISCUSSION THREAD

**Discussion by panawala** [alibaba/ilogtail#236] [Help] [answered]
麻烦提供下ilogtail测试框架的step by step用法？

**Comment by henryzhx8** [comment, marked as answer]
在大部分情况下，您只需要编写一个Yaml配置文件来定义测试行为，即可轻松完成测试。如果测试流程涉及外部环境依赖，您只需额外提供该环境的镜像（或构建镜像所需文件），从而省却了人工配置测试环境（包括网络等）的麻烦。

针对您开发插件的类型不同，测试引擎为您提供了不同的组件方便您生成所需的日志或验证日志内容，具体步骤可参见[E2E测试流程](https://github.com/alibaba/ilogtail/blob/main/docs/cn/developer-guide/test/e2e-test.md)。
